R/then.R

```diff
@@ -91,23 +105,29 @@
 #'
 #' @param promise A promise object. The object can be in any state.
 #'
-#' @param onFulfilled A function (or a formula--see Details) that will be
-#'   invoked if the promise value successfully resolves. When invoked, the
-#'   function will be called with a single argument: the resolved value.
-#'   Optionally, the function can take a second parameter `.visible` if you care
-#'   whether the promise was resolved with a visible or invisible value. The
-#'   function can return a value or a promise object, or can throw an error;
-#'   these will affect the resolution of the promise object that is returned
-#'   by `then()`.
-#'
-#' @param onRejected A function taking the argument `error` (or a formula--see
-#'   Details). The function can return a value or a promise object, or can throw
-#'   an error. If `onRejected` is provided and doesn't throw an error (or return
-#'   a promise that fails) then this is the async equivalent of catching an
-#'   error.
-#'
+#' @param onFulfilled A function that will be invoked if the promise value
+#'   successfully resolves. When invoked, the function will be called with a
+#'   single argument: the resolved value. Optionally, the function can take a
+#'   second parameter `.visible` if you care whether the promise was resolved
+#'   with a visible or invisible value. The function can return a value or a
+#'   promise object, or can throw an error; these will affect the resolution of
+#'   the promise object that is returned by `then()`.
+#'
+#' @param onRejected A function taking the argument `error`. The function can
+#'   return a value or a promise object, or can throw an error. If `onRejected`
+#'   is provided and doesn't throw an error (or return a promise that fails)
+#'   then this is the async equivalent of catching an error.
+#' @param ... Ignored.
 #' @export
-then <- function(promise, onFulfilled = NULL, onRejected = NULL) {
+then <- function(
+  promise,
+  onFulfilled = NULL,
+  onRejected = NULL,
+  ...,
+  tee = FALSE
+) {
+  rlang::check_dots_empty()
```

Comment:
If we had the type checkers standalone helper, I'd recommend checking `tee` here

```
check_bool(tee)
```

Using the type checkers could be a future maintenance thing, but I still think we should check or validate tee, maybe even just `tee <- isTRUE(tee)`